File: test/initial/ContactViewModel.cs
using System.Collections.Generic;

namespace SSI.Domain.ViewModels
{
    public class ContactViewModel
    {
        public int Id { get; set; }
        
        public int TiersId { get; set; }
        
        public EntityItemViewModel Titre { get; set; }
        
        public string Prenom { get; set; }
        
        public string Nom { get; set; }
        
        public bool IsFormateur { get; set; }
        
        public EntityItemViewModel Fonction { get; set; }
        
        public EntityItemViewModel Service { get; set; }
        
        public string Telephone { get; set; }
        
        public string Mobile { get; set; }
        
        public string Email { get; set; }
        
        public string Descriptif { get; set; }

        public bool IsActif { get; set; }

        //public List<EntityItemViewModel> AllTitres { get; set; }
        //public List<EntityItemViewModel> AllFonctions { get; set; }
        //public List<EntityItemViewModel> AllServices { get; set; }
        //public bool IsEditMode { get; set; }

        //[Display(Name = "Nom complet")]
        //public string FullName {
        //    get {
        //        return $"{Prenom} {Nom}";
        //    }
        //}
    }
}

File: test/final/ContactViewModel.js
/// <reference path="knockout.d.ts"/>
var ContactViewModel = (function () {
    function ContactViewModel() {
        this.Id = ko.observable(0);
        this.TiersId = ko.observable(0);
        this.Titre = ko.observable(new EntityItemViewModel());
        this.Prenom = ko.observable("");
        this.Nom = ko.observable("");
        this.IsFormateur = ko.observable(false);
        this.Fonction = ko.observable(new EntityItemViewModel());
        this.Service = ko.observable(new EntityItemViewModel());
        this.Telephone = ko.observable("");
        this.Mobile = ko.observable("");
        this.Email = ko.observable("");
        this.Descriptif = ko.observable("");
        this.IsActif = ko.observable(false);
        this.AllTitres = ko.observableArray([]);
        this.AllFonctions = ko.observableArray([]);
        this.AllServices = ko.observableArray([]);
        this.IsEditMode = ko.observable(false);
    }
    return ContactViewModel;
})();
